semgrep/semgrep/cli.py

```diff
@@ -233,48 +232,43 @@ def cli() -> None:
     # change cwd if using docker
     semgrep.config_resolver.adjust_for_docker(args.precommit)
 
-    try:
-        if args.dump_ast:
-            if not args.lang:
-                print_error_exit("language must be specified to dump ASTs")
-            else:
-                dump_parsed_ast(args.json, args.lang, args.pattern, args.target)
-        elif args.validate:
-            _, invalid_configs = semgrep.semgrep_main.get_config(
-                args.generate_config, args.pattern, args.lang, args.config
-            )
-            if invalid_configs:
-                print_error_exit(
-                    f"run with --validate and there were {len(invalid_configs)} errors loading configs"
-                )
-            else:
-                print_error("Config is valid")
-
-        elif args.test:
-            semgrep.test.test_main(args)
+    if args.dump_ast:
+        if not args.lang:
+            print_error_exit("language must be specified to dump ASTs")
         else:
-            semgrep.semgrep_main.main(
-                target=args.target,
-                pattern=args.pattern,
-                lang=args.lang,
-                config=args.config,
-                generate_config=args.generate_config,
-                no_rewrite_rule_ids=args.no_rewrite_rule_ids,
-                jobs=args.jobs,
-                include=args.include,
-                include_dir=args.include_dir,
-                exclude=args.exclude,
-                exclude_dir=args.exclude_dir,
-                json_format=args.json,
-                sarif=args.sarif,
-                output_destination=args.output,
-                quiet=args.quiet,
-                strict=args.strict,
-                exit_on_error=args.error,
-                autofix=args.autofix,
-                dangerously_allow_arbitrary_code_execution_from_rules=args.dangerously_allow_arbitrary_code_execution_from_rules,
+            dump_parsed_ast(args.json, args.lang, args.pattern, args.target)
+    elif args.validate:
+        _, invalid_configs = semgrep.semgrep_main.get_config(
+            args.generate_config, args.pattern, args.lang, args.config
+        )
+        if invalid_configs:
+            print_error_exit(
+                f"run with --validate and there were {len(invalid_configs)} errors loading configs"
             )
-    except NotImplementedError as ex:
```

Comment:
We were raising `NotImplementedError` in a few different places, but I don't believe that's the intended use of `NotImplementedError`. It's mostly for raising in a base class function that should never be called directly, but instead via a child class implementation, not for normal unimplemented functionality:

```python
In [21]: NotImplementedError?                                                                           
Init signature: NotImplementedError(self, /, *args, **kwargs)
Docstring:      Method or function hasn't been implemented yet.
Type:           type
Subclasses:     StdinNotImplementedError
```

It's better to catch our own internally defined errors here anyway rather than re-purposing `NotImplementedError`. And finally, we can totally remove this `try`/`except` because the only place the `cli` function is called is in `__main__` and it's wrapped in a `SemgrepError` `try`/`except` there.